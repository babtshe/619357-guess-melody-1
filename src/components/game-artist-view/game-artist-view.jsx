import React from 'react';
import PropTypes from 'prop-types';

const GameArtistView = (props) => {
  const {audioSrc, answers, onAnswer} = props;
  return (
    <section className="game game--artist">
      <header className="game__header">
        <a className="game__back" href="#">
          <span className="visually-hidden">Сыграть ещё раз</span>
          <img className="game__logo" src="img/melody-logo-ginger.png" alt="Угадай мелодию"/>
        </a>

        <svg xmlns="http://www.w3.org/2000/svg" className="timer" viewBox="0 0 780 780">
          <circle className="timer__line" cx="390" cy="390" r="370" style={{
            filter: `url(#blur)`,
            transform: `rotate(-90deg) scaleY(-1)`,
            transformOrigin: `center`
          }} />
        </svg>

        <div className="timer__value" xmlns="http://www.w3.org/1999/xhtml">
          <span className="timer__mins">05</span>
          <span className="timer__dots">:</span>
          <span className="timer__secs">00</span>
        </div>

        <div className="game__mistakes">
          <div className="wrong"></div>
          <div className="wrong"></div>
          <div className="wrong"></div>
        </div>
      </header>

      <section className="game__screen">
        <h2 className="game__title">Кто исполняет эту песню?</h2>
        <div className="game__track">
          <button className="track__button track__button--play" type="button"></button>
          <audio src={audioSrc} autoPlay></audio>
        </div>

        <form className="game__artist" onChange={ (evt)=> {
          evt.preventDefault();
          onAnswer();
        }}>
          {answers.map((answer, index) => {
            return (
              <div className="artist" key={`artist-${index}`}>
                <input className="artist__input visually-hidden" type="radio" name="answer" value={`artist-${index}`} id={`answer-${index}`}/>
                <label className="artist__name" htmlFor={`answer-${index}`}>
                  <img className="artist__picture" src={answer.imageSrc} alt={answer.name}/>
                  {answer.name}
                </label>
              </div>
            );
          })}
        </form>
      </section>
    </section>
  );
};

GameArtistView.propTypes = {
  audioSrc: PropTypes.string.isRequired,
  answers: PropTypes.arrayOf(PropTypes.shape({
    imageSrc: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  })).isRequired,
  onAnswer: PropTypes.func.isRequired,
};

export {GameArtistView};
