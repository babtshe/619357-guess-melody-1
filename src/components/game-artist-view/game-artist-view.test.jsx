import React from 'react';
import renderer from 'react-test-renderer';
import {GameArtistView} from './game-artist-view';

const mock = {
  question: {
    audioSrc: `test.file`,
    artist: `Test Artist`,
    answers: [
      {
        imageSrc: `test.file-a`,
        name: `Artist A`,
      },
      {
        imageSrc: `test.file-b`,
        name: `Artist B`,
      },
      {
        imageSrc: `test.file-c`,
        name: `Artist C`,
      },
      {
        imageSrc: `test.file-d`,
        name: `Artist D`,
      },
    ]
  }
};

it(`GameGenreView is rendering correctly`, ()=> {
  const {question} = mock;
  const tree = renderer.create(<GameArtistView
    onAnswer = {jest.fn()}
    audioSrc = {question.audioSrc}
    artist = {question.artist}
    answers = {question.answers}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
