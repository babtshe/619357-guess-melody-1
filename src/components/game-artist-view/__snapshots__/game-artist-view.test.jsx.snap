// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameGenreView is rendering correctly 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <audio
        autoPlay={true}
        src="test.file"
      />
    </div>
    <form
      className="game__artist"
      onChange={[Function]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-0"
          name="answer"
          type="radio"
          value="artist-0"
        />
        <label
          className="artist__name"
          htmlFor="answer-0"
        >
          <img
            alt="Artist A"
            className="artist__picture"
            src="test.file-a"
          />
          Artist A
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          type="radio"
          value="artist-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Artist B"
            className="artist__picture"
            src="test.file-b"
          />
          Artist B
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-2"
          name="answer"
          type="radio"
          value="artist-2"
        />
        <label
          className="artist__name"
          htmlFor="answer-2"
        >
          <img
            alt="Artist C"
            className="artist__picture"
            src="test.file-c"
          />
          Artist C
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-3"
          name="answer"
          type="radio"
          value="artist-3"
        />
        <label
          className="artist__name"
          htmlFor="answer-3"
        >
          <img
            alt="Artist D"
            className="artist__picture"
            src="test.file-d"
          />
          Artist D
        </label>
      </div>
    </form>
  </section>
</section>
`;
